### PR TITLE
ref(theme): Decrease input & button heights

### PR DESCRIPTION
**Pull request by vuluongj20**
Decrease the height of form inputs and buttons by 2 pixels (for all sizes). This is warranted as we recently changed the default font size from `16px` down to `14px`.

A number of styled input/button components have hard-coded heights (`40px`/`41px`). They've been updated with the standardized height values from `theme` (see screenshots in the comments).

**Before:**
image

**After:**
image

### CODE REVIEW

File: static/app/components/pageTimeRangeSelector.tsx [hunk 1]
```diff
@@ -27,7 +27,7 @@ const DropdownDate = styled(Panel)`
   display: flex;
   justify-content: center;
   align-items: center;
-  height: 42px;
+  height: ${p => p.theme.form.md.height}px;
```
Comment (vuluongj20) [inline static/app/components/pageTimeRangeSelector.tsx:30]: Date Range Last 14 days

File: static/app/components/smartSearchBar/index.tsx [hunk 2]
```diff
@@ -1899,10 +1899,11 @@ export default withApi(withRouter(withOrganization(SmartSearchBarContainer)));
 export {SmartSearchBar, Props as SmartSearchBarProps};
 
 const Container = styled('div')<{inputHasFocus: boolean}>`
+  height: ${p => p.theme.form.md.height}px;
   border: 1px solid ${p => p.theme.border};
   box-shadow: inset ${p => p.theme.dropShadowLight};
   background: ${p => p.theme.background};
-  padding: 7px ${space(1)};
+  padding: 6px ${space(1)};
```
Comment (vuluongj20) [inline static/app/components/smartSearchBar/index.tsx:1906]: Screen Shot 2022-10-17 at 1 36 20 PM

File: static/app/views/dashboardsV2/widgetBuilder/buildSteps/groupByStep/queryField.tsx [hunk 3]
```diff
@@ -75,7 +75,7 @@ export function QueryField({
 }
 
 const DragAndReorderButton = styled(Button)`
-  height: 40px;
+  height: ${p => p.theme.form.md.height}px;
```
Comment (vuluongj20) [inline static/app/views/dashboardsV2/widgetBuilder/buildSteps/groupByStep/queryField.tsx:78]: Screen Shot 2022-10-17 at 12 28 03 PM

File: static/app/views/dashboardsV2/widgetBuilder/buildSteps/groupByStep/sortableQueryField.tsx [hunk 4]
```diff
@@ -26,7 +26,7 @@ export function SortableQueryField({dragId, ...props}: SortableItemProps) {
     style = {
       ...style,
       zIndex: 100,
-      height: '40px',
+      height: theme.form.md.height,
```
Comment (vuluongj20) [inline static/app/views/dashboardsV2/widgetBuilder/buildSteps/groupByStep/sortableQueryField.tsx:29]: timestamp

File: static/app/views/eventsV2/savedQuery/index.tsx [hunk 5]
```diff
@@ -530,7 +530,6 @@ const SaveAsButton = styled(Button)`
 `;
 
 const SaveAsInput = styled(InputControl)`
-  height: 40px;
   margin-bottom: ${space(1)};
 `;
```
Comment (vuluongj20) [inline static/app/views/eventsV2/savedQuery/index.tsx:534]: Display name

File: static/app/views/eventsV2/table/arithmeticInput.tsx [hunk 6]
```diff
@@ -285,8 +285,6 @@ export default class ArithmeticInput extends PureComponent<Props, State> {
 }
 
 const Container = styled('div')<{isOpen: boolean}>`
-  border: 1px solid ${p => p.theme.border};
-  box-shadow: inset ${p => p.theme.dropShadowLight};
   background: ${p => p.theme.background};
   position: relative;
```
Comment (vuluongj20) [inline static/app/views/eventsV2/table/arithmeticInput.tsx:290]: **Before:**
count()

**After:**
count()

File: static/app/views/eventsV2/table/columnEditCollection.tsx [hunk 7]
```diff
@@ -691,7 +691,7 @@ const DragPlaceholder = styled('div')`
   margin: 0 ${space(3)} ${space(1)} ${space(3)};
   border: 2px dashed ${p => p.theme.border};
   border-radius: ${p => p.theme.borderRadius};
-  height: 41px;
+  height: ${p => p.theme.form.md.height}px;
```
Comment (vuluongj20) [inline static/app/views/eventsV2/table/columnEditCollection.tsx:694]: Screen Shot 2022-10-17 at 12 43 42 PM

File: static/app/views/eventsV2/table/columnEditCollection.tsx [hunk 8]
```diff
@@ -728,11 +726,11 @@ const AliasField = styled('div')<{singleColumn: boolean}>`
...
 const RemoveButton = styled(Button)`
   margin-left: ${space(1)};
-  height: 40px;
+  height: ${p => p.theme.form.md.height}px;
 `;
 
 const DragAndReorderButton = styled(Button)`
-  height: 40px;
+  height: ${p => p.theme.form.md.height}px;
 `;
```
Comment (vuluongj20) [inline static/app/views/eventsV2/table/columnEditCollection.tsx:734]: (Required)

File: static/app/views/eventsV2/table/queryField.tsx [hunk 9]
```diff
@@ -814,14 +814,12 @@ class BufferedInput extends Component<BufferedInputProps, InputState> {
...
-  /* Match the height of the select boxes */
-  height: 41px;
   min-width: 50px;
 `;
 
 const BlankSpace = styled('div')`
   /* Match the height of the select boxes */
-  height: 41px;
+  height: ${p => p.theme.form.md.height}px;
   min-width: 50px;
```
Comment (vuluongj20) [inline static/app/views/eventsV2/table/queryField.tsx:823]: Screen Shot 2022-10-17 at 1 19 54 PM

File: static/app/views/organizationStats/teamInsights/controls.tsx [hunk 10]
```diff
@@ -247,8 +247,6 @@ const StyledTeamSelector = styled(TeamSelector)`
 `;
 
 const StyledPageTimeRangeSelector = styled(PageTimeRangeSelector)`
-  height: 40px;
-
   div {
     min-height: unset;
   }
```
Comment (vuluongj20) [inline static/app/views/organizationStats/teamInsights/controls.tsx:252]: Date Range Last 14 days

File: static/app/views/releases/detail/overview/index.tsx [hunk 11]
```diff
@@ -458,7 +458,7 @@ class ReleaseOverview extends AsyncView<Props> {
                           )}
                           <ReleaseDetailsPageFilters>
                             <EnvironmentPageFilter />
-                            <StyledPageTimeRangeSelector
+                            <PageTimeRangeSelector
```
Comment (vuluongj20) [inline static/app/views/releases/detail/overview/index.tsx:461]: `PageTimeRangeSelector` now matches button/form input heights, so there's no need to re-style it.
Crash Free Session F

File: static/app/views/settings/components/dataScrubbing/modals/form/eventIdField.tsx [hunk 12]
```diff
@@ -155,7 +155,7 @@ const StyledInput = styled(Input)`
 `;
 
 const Status = styled('div')`
-  height: 40px;
+  height: 100%;
   position: absolute;
```
Comment (vuluongj20) [inline static/app/views/settings/components/dataScrubbing/modals/form/eventIdField.tsx:159]: Event ID (Optional) ®

File: static/app/views/settings/components/dataScrubbing/modals/form/selectField.tsx [hunk 13]
```diff
@@ -32,13 +32,6 @@ class SelectField extends Component<Props> {
...
         }))}
-        styles={{
-          control: (provided: {[x: string]: string | number | boolean}) => ({
-            ...provided,
-            minHeight: '41px',
-            height: '41px',
-          }),
-        }}
         ref={this.selectRef}
         openOnFocus
```
Comment (vuluongj20) [inline static/app/views/settings/components/dataScrubbing/modals/form/selectField.tsx:36]: Note sure why we needed `41px` here
Credit card numbers

File: static/app/views/settings/project/projectOwnership/ruleBuilder.tsx [hunk 14]
```diff
@@ -228,7 +228,6 @@ const StyledIconAdd = styled(IconAdd)`
 
 const BuilderBar = styled('div')`
   display: flex;
-  height: 40px;
   align-items: center;
   margin-bottom: ${space(2)};
 `;
```
Comment (vuluongj20) [inline static/app/views/settings/project/projectOwnership/ruleBuilder.tsx:233]: `BuilderBar` doesn't need an explicit height, it'll just expand to accommodate its children.
srcexample

**Review by malwilley — APPROVED** [review]
Looks good to me - thanks for including screenshots with all the additional changes you made 😄

**Pull request merged**